airbyte-integrations/connectors/source-bamboo-hr/source_bamboo_hr/source.py

```diff
@@ -26,146 +42,441 @@ def __init__(self, config: Mapping[str, Any]) -> None:
+            ):
+                raise e
+
+            # Otherwise, just log a warning.
+            self.logger.warning(
+                f"Stream `{self.name}`. An error occurred, details: {e}. Skipping for now."
+            )
+            yield {}
 
-    def _get_json_schema_from_file(self):
-        return super().get_json_schema()
 
-    @staticmethod
-    def _union_schemas(schema1, schema2):
-        schema1["properties"] = {**schema1["properties"], **schema2["properties"]}
-        return schema1
+class CustomReportsStream(BambooHrStream):
+    primary_key = None # type: ignore
 
-    def get_json_schema(self) -> Mapping[str, Any]:
-        """
-        Returns the JSON schema.
+    def __init__(self, *args, **kwargs): # type: ignore
+        super().__init__(*args, **kwargs) # type: ignore
 
-        The final schema is constructed by first generating a schema for the fields
-        in the config and, if default fields should be included, adding these to the
-        schema.
-        """
-        schema = self._get_json_schema_from_config()
-        if self.config.get("custom_reports_include_default_fields"):
-            default_schema = self._get_json_schema_from_file()
-            schema = self._union_schemas(default_schema, schema)
-        return schema
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]: # type: ignore
+        available_fields: List[str] = self.config.get("available_fields", [])
+        for fields in chunk_iterable(available_fields, 100):
+            yield {"fields": fields}
 
-    def path(self, **kwargs) -> str:
+    def path(self, **kwargs) -> str: # type: ignore
         return "reports/custom"
 
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] | None = None,
+        stream_slice: Mapping[str, Any] | None = None,
+        stream_state: Mapping[str, Any] | None = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        for record in super().read_records(
+            sync_mode, cursor_field, stream_slice, stream_state
+        ):
+            # Augment the record with the table name.
+            if record == {} or not isinstance(record,Mapping):
+                self.logger.warn("Empty record or non-map record encountered in CustomReportsStream.")
+                continue
+            else:
+                new_record = {
+                    "id": record["id"],
+                    "data": record,
+                }
+                yield new_record
+
     @property
     def http_method(self) -> str:
         return "POST"
 
-    def request_body_json(self, **kwargs) -> Optional[Mapping]:
-        return {"title": "Airbyte", "fields": list(self.schema["properties"].keys())}
+    @staticmethod
+    def get_default_bamboo_fields() -> List[str]:
+        # As per https://documentation.bamboohr.com/docs/list-of-field-names
+        return [
+            "acaStatus"
```

Comment:
Let's remove this as it's deprecated

Reply:
My only concern is we may encounter some client with a legacy setup that is still making use of it (and not the new version, acaStatusCategory).  Seems like we might want to err on the side of caution and just try to pull it in.  We can choose to use it (or not) at the transform layer.  What do you think?

Reply:
should be fine, I think we just need to check that the deprecated fields don't result in weird bamboo endpoint error. I rmb previously when we included a deprecated field `flsaCode` it basically shifted all the values and keys and caused the values to be on the wrong keys in the response.

Reply:
Got it, I'll remove then.  We can always add it back if someone complains (unlikely).